README.md

```diff
@@ -1,95 +1,149 @@
+    project-name:
+      lang: rust
+mygitea:
+  myorg2:
+    project-name2:
+      lang: python
 ```
-* These results are stored in a Moneta store under a key generated by `RepositoryGem#identifier` (for e.g. a `GithubGem`, the identifier will be of the form `'github/org/name'`).
-* The Moneta store used by the Worker is provided by the Sinatra app Class: `::CuttingEdge::App.store`.
-  * So we must do e.g. `CuttingEdge::App.set(:store, Moneta.new(:Memory))` to set the `#store` method on the `CuttingEdge` Class.
 
-### Example use of the DependencyWorker and Gem API
+Don't forget to run CuttingEdge with the `--config` option!
+
+### Authorization and private repositories
 
-See `test.rb` for a basic example (without Sinatra) of how the setup works. Run it with `bundle exec ruby test.rb`.
+If you want to monitor dependencies in a private (e.g. GitHub or Gitlab) project, you can instruct CuttingEdge to use an API token for accessing the dependency files. In `projects.yml`:
+
+```yaml
+github:
+   secret-org:
+     secret-repo:
+       auth_token: 'mysecrettoken'
+```
+
+For info on geneating API tokens, see:
+
+* [GitHub](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token)
+* [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
+* [Gitea](https://docs.gitea.io/en-us/api-usage/)
+
+If you don't want to expose your API token in `project.yml` (**which you shouldn't** when it is publicly accessible on the internet), you can instead define your project repository programatically in `config.rb`. This will allow you to set secrets (auth_token, but if desired also name and org of the repository) by (for instance) utilising environment variables:
 
 ```ruby
-require 'redis-objects'
+module CuttingEdge
+    require './lib/cutting_edge/repo.rb'
+    REPOSITORIES = {
+      "gitlab/#{ENV['SECRET_REPO1_ORG']}/#{ENV['SECRET_REPO1_NAME']}" => GitlabRepository.new(org: ENV['SECRET_REPO1_ORG'], name: ENV['SECRET_REPO1_NAME'], auth_token: ENV['SECRET_REPO1_TOKEN'], hide: true)
+    }
+end
+```
 
-gem = GithubGem.new('gollum', 'gollum')
+Setting the `hide` key to true will ensure your private repo is not listed in the web frontend, unless you enter your CuttingEdge secret token on the landing page.
 
-Redis::Objects.redis = Redis.new
-gem_dependencies = Redis::Value.new(gem.identifier)
+This approach is especially useful on Heroku, where you can use [Heroku config variables](https://devcenter.heroku.com/articles/config-vars).
 
-puts gem_dependencies.value # Currently nil
+# Refreshing dependency status through git hooks
 
-DependencyWorker.perform_async(gem.identifier, gem.gemspec_location, gem.gemfile_location) # Fire up a Sidekiq job
+CuttingEdge by default checks whether the status of your dependencies has changed once every hour. However, if you wish, you can also setup hooks so that dependency status is checked (for instance) whenever you make a commit to your project.
```

Comment:
This is cool. Have we tested this yet?

Reply:
No, although the`refresh` route is tested. Once we have an up to date server running on Heroku, I'll test it with gollum!